### PR TITLE
Add Mypy to GitHub Actions and 'pre-commit' configuration

### PR DESCRIPTION
**Pull request by youngyojun**
This revision includes:

- [x] 'pre-commit' configuration for 'mypy'
- [x] GitHub Actions for 'mypy'

### CODE REVIEW

File: src/polytope/github/repository/Repository.py [hunk 1]
```diff
@@ -9,14 +9,12 @@
 from polytope.github.RequestVerb import RequestVerb
 from polytope.github.Session import RequestsSession
 from polytope.github.Session import Session
+from polytope.github.Token import Token
```
Comment (TAMREF) [inline src/polytope/github/repository/Repository.py:12]: Now can we replace `"Token"` type from initializer of Requester and Repository with `Token`?

**Review by TAMREF — APPROVED** [review]
LGTM. Thank you for the work!
I'll get back when github action is ready.

**Review by TAMREF — APPROVED** [review]
Thanks for the work. LGTM!

File: src/polytope/github/repository/Repository.py (hunk 1 above)
Comment (youngyojun) [reply src/polytope/github/repository/Repository.py:12]: Yes, we can. Thanks for your pointing out!
I will replace all `"Token"` with `Token`.

**Review by utilForever — COMMENTED** [review]
Good! 🤗

**Pull request merged**